libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteInterpolator.kt

```diff
@@ -31,35 +33,164 @@ internal class ReplayRouteInterpolator {
 
     /**
      * Interpolate a speed across a distance. The returned segment will include approximate
-     * once per second speed and distance calculations that can be used to create location coordinates.
+     * speed and distance calculations that can be used to create location coordinates.
+     * This will also calculate and interpolate cruising speeds, which allows for the start and
+     * end speeds to be zero.
      */
     fun interpolateSpeed(
         options: ReplayRouteOptions,
         startSpeed: Double,
         endSpeed: Double,
         distance: Double
     ): ReplayRouteSegment {
+        val segment = calculateSegmentSpeedAndDistances(options, startSpeed, endSpeed, distance)
         val speedSteps = mutableListOf<ReplayRouteStep>()
-        speedSteps.add(
-            ReplayRouteStep(
-                acceleration = 0.0,
-                speedMps = startSpeed,
-                positionMeters = 0.0
-            )
+
+        // Add the speeding up steps, if there are any
```

Comment:
NIT maybe move into a `private fun`ction with a name based on the comment so comment wouldn't be necessary? 🤔

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/5724#discussion_r868710426